### PR TITLE
gradle: 7.0 -> 7.3

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
Brings us one step closer to #301

### CODE REVIEW

**Comment by lorenzleutgeb** [comment]
Happy to port this to `modules`.

File: build.gradle [hunk 1]
```diff
@@ -107,8 +107,8 @@ task bundledJar(type: Jar) {
 
 jacocoTestReport {
 	reports {
-		xml.enabled = true
-		html.enabled = true
+		xml.required = true
```
Comment (lorenzleutgeb) [inline build.gradle:110]: To be compatible with future releases, see [the docs](https://docs.gradle.org/7.3/dsl/org.gradle.api.reporting.Report.html#org.gradle.api.reporting.Report:enabled).

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -3,10 +3,8 @@ on: [push, pull_request]
 jobs:
   build:
     runs-on: ${{ matrix.os }}-latest
-    continue-on-error: ${{ matrix.continue-on-error }}
     strategy:
       matrix:
-        continue-on-error: [false]
```
Comment (lorenzleutgeb) [inline .github/workflows/build.yml:9]: Wasn't used anyway, and now we have multiple stable Java versions. Can re-add if we decide to include an "Early Access Preview" release with unstable CI.

**Review by AntoniusW — APPROVED** [review]
Looks good to me.

File: gradlew [hunk 3]
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (lorenzleutgeb) [inline gradlew:1]: Changes in this file were automatically done by Gradle.

**Review by madmike200590 — APPROVED** [review]
Looks good to me!

**Comment by madmike200590** [comment]
> Happy to port this to `modules`.

That would be great, thanks!

**Pull request merged**